src/test/java/com/zaxxer/nuprocess/DirectWriteTest.java

```diff
@@ -105,7 +107,9 @@ public void onStdout(ByteBuffer buffer, boolean closed)
 
       NuProcessBuilder pb = new NuProcessBuilder(processListener, command);
       NuProcess nuProcess = pb.start();
-      for (int i = 0; i < 1000; i++) {
+      // TODO: given a large i (e.g. 1,000, 10,000), this unit test (testConsecutiveWrites) will
+      //       produce a side-effect on InterruptTest (has problem on Mac OS X, but works on Linux and Win32).
+      for (int i = 0; i < 100; i++) {
```

Comment:
We may be able to avoid the side-effect by [configuring the surefire](https://github.com/brettwooldridge/HikariCP/blob/dev/pom.xml#L294) plugin in this project's pom.xml and setting it to not reuse forks (``forkCount=1/reuseForks=true`` documentation [here](https://maven.apache.org/surefire/maven-surefire-plugin/examples/fork-options-and-parallel-execution.html)).  

Worth a try.